main.js

```diff
@@ -0,0 +1,266 @@
+const startingBoard = function() {
+  return [[5, 4, 3, 2, 1],[],[]];
+};
+
+const printBoard = function() {
+  this.board.map(function(peg) {
+    let output = "---";
+    peg.forEach(function(disc) {
+      output += ` ${disc} `;
+    })
+    console.log(output);
+    logConsole(output);
+  })
+};
+
+const startBoard = function() {
+  // Make a copy of the startingBoard so that it isn't mutated.
+  this.board = startingBoard().map(function (item) {
+    return item;
+  });
+  boardState.update();
+  boardState.print();
+  boardState.check();
+};
+
+const updatePegs = function() {
+  // Establishes references to the items in the board array.
+  // Mutation of the peg means mutation of the board property.
+  this.peg1 = this.board[0];
+  this.peg2 = this.board[1];
+  this.peg3 = this.board[2]; 
+}
+
+const moveDisc = function(source, destination) {
+  clearStatus();
+  
+  let sourcePeg = this[`peg${source}`];
+  let destinationPeg = this[`peg${destination}`];
+
+  // What if I try to move a disc from a peg that is empty?
+  if (sourcePeg.length === 0) {
+    let message = "Sorry. You can't move a disc that doesn't exist!";
+    console.log(message);
+    clearConsole();
+    displayStatus(message);
+    boardState.print();
+    this.moveCount += 1;
+    clearMoves();
+    displayMoves(this.moveCount);
+    return;
+  }
+
+  // What if I try to move a larger disc on top of a smaller one?
+  if (sourcePeg[sourcePeg.length - 1] > destinationPeg[destinationPeg.length - 1]) {
+    let message = "Sorry. You can't move a larger disc on top of a smaller disc.";
+    console.log(message)
+    clearConsole();
+    displayStatus(message);
+    boardState.print();
+    this.moveCount += 1;
+    clearMoves();
+    displayMoves(this.moveCount);
+    return;
+  }
+
+  // What if I try to move the disc to the same peg?
+  if (sourcePeg === destinationPeg) {
+    let message = "Nothing changed...";
+    console.log(message);
+    clearConsole();
+    displayStatus(message);
+    boardState.print();
+    this.moveCount += 1;
+    clearMoves();
+    displayMoves(this.moveCount);
+    return;
+  }
+
+  let disc = sourcePeg.pop();
+  destinationPeg.push(disc);
+  clearConsole();
+  boardState.print();
+  clearMoves();
+  this.moveCount += 1;
+  displayMoves(this.moveCount);
+  boardState.check();
+};
+
+const resetBoard = function() {
+  console.log("Resetting game...")
+  clearStatus();
+  clearConsole();
+  clearMoves();
+  hideButton('reset');
+  hideButton('noReset');
+  boardState.moveCount = 0
+  boardState.start();
+};
+
+const checkWinner = function() {
+  // What is the winning condition?
+  // When all of the discs are not on 'peg1' and the discs are in their original order
+  // let winningCondition = [5, 4, 3, 2, 1];
+  
+  // Put the pegs in an array
+  let pattern = /peg/;
+  let pegs = Object.keys(boardState).filter(function(key) {
+    if (pattern.test(key)) {
+      return key;
+    }
+  });
+
+  // Is peg1 empty?
+  if (this.peg1.length === 0) {

```

Comment:
this is the line where it breaks cause there is no reference to this when you call checkWinner = you create a variable called pegs above, you could check length on this